okhttp/src/main/java/okhttp3/internal/ws/MessageInflater.kt

```diff
@@ -0,0 +1,36 @@
+package okhttp3.internal.ws
+
+import okio.Buffer
+import okio.InflaterSource
+import java.io.Closeable
+import java.io.EOFException
+import java.io.IOException
+import java.util.zip.Inflater
+
+class MessageInflater : Closeable {
+  private val source = Buffer()
+  private val inflater = Inflater(true /* omit zlib header */)
+  private val inflaterSource = InflaterSource(source, inflater)
+
+  private var totalBytesToInflate = 0L
+
+  @Throws(IOException::class)
+  fun inflate(deflated: Buffer): Buffer {
+    source.writeAll(deflated)
+    totalBytesToInflate += source.size
+
+    return Buffer().apply {
```

Comment:
should this method write the inflated bytes back to the caller’s buffer?

Reply:
It’s already destructive and that saves an allocation

Reply:
Done